api/server/middleware/middleware.go

```diff
@@ -112,6 +114,40 @@ func AcceptHeaderHandler(serverAcceptedTypes []string) Middleware {
 	}
 }
 
+func GzipMiddleware() Middleware {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request){
+			if !strings.Contains(r.Header.Get("Accept-Encoding"), "gzip") {
+				next.ServeHTTP(w, r)
+				return
+		}
+
+		w.Header().Set("Content-Encoding", "gzip")
+
+		gz := gzip.NewWriter(w)
+		// defer gz.Close()
+		defer func(){
+			if err:= gz.Close(); err != nil {
+				log.WithError(err).Error("failed to close gzip writer")
```

Comment:
```suggestion
				log.WithError(err).Error("Failed to close gzip writer")
```
We use upper casing for logs